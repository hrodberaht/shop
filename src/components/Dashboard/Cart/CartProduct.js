import React, { Component } from 'react';
import PropTypes from 'prop-types';

export default class CartProduct extends Component {
  render() {
    const {
      product: {
        name, price, pcsOrder, totalPrice,
      },
    } = this.props;
    return (
      <tr>
        <td>{name}</td>
        <td>{price}</td>
        <td>{pcsOrder}</td>
        <td>
          <span>$</span>
          {totalPrice}
        </td>
      </tr>
    );
  }
}

CartProduct.propTypes = {
  product: PropTypes.shape({
    name: PropTypes.string,
    price: PropTypes.number,
    pcsOrder: PropTypes.number,
    totalPrice: PropTypes.number,
  }).isRequired,
};
