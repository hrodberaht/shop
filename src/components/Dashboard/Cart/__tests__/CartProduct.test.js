import CartProduct from '../CartProduct';

const defaultProps = {
  product: {
    id: '1234',
    name: 'fx',
    price: 400,
    pcsOrder: 2,
    totalPrice: 800,
  },
  productAfterUpdate: {
    id: '1234',
    name: 'fx',
    price: 400,
    pcsOrder: 3,
    totalPrice: 1200,
  },
  remove: jest.fn(),
  changeQuantity: jest.fn(),
};
const setup = buildSetup(CartProduct, defaultProps);

describe('<CartProduct />', () => {
  it('should render witout crash', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });

  it('state quantity should be equal to product pcsOrder', () => {
    const { wrapper, props } = setup();
    expect(wrapper.state().quantity).toBe(props.product.pcsOrder);
  });
  it('should call remove after click', () => {
    const { wrapper } = setup();
    wrapper.find('button').simulate('click');
    expect(wrapper.instance().props.remove).toHaveBeenCalled();
  });
  it('should change state if input value change', () => {
    const { wrapper } = setup();
    wrapper.find('input').simulate('change', { target: { value: '3' } });
    expect(wrapper.state().quantity).toBe(3);
  });

  it('should call changeQuantit with productAfterUpdate if input value change', () => {
    const { wrapper, props } = setup();
    wrapper.find('input').simulate('change', { target: { value: '3' } });
    expect(wrapper.instance().props.changeQuantity).toHaveBeenCalledWith(props.productAfterUpdate);
  });
  it('state should be 1 if input value is < 1 and ', () => {
    const { wrapper } = setup();
    wrapper.find('input').simulate('change', { target: { value: '-1' } });
    expect(wrapper.state().quantity).toBe(1);
  });
});
