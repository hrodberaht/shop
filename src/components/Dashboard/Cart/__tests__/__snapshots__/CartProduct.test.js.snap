// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CartProduct /> should render witout crash 1`] = `
<tr>
  <td>
    fx
  </td>
  <td>
    400
  </td>
  <td>
    <input
      onChange={[Function]}
      type="number"
      value={2}
    />
  </td>
  <td>
    <span>
      $
    </span>
    800
  </td>
  <td>
    <button
      className="btn btn-danger"
      onClick={[Function]}
      type="button"
    >
      X
    </button>
  </td>
</tr>
`;
